src/catalog/src/durable/upgrade/stash/v44_to_v45.rs

```diff
@@ -0,0 +1,323 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use mz_stash::upgrade::{wire_compatible, MigrationAction};
+use mz_stash::{Transaction, TypedCollection};
+use mz_stash_types::StashError;
+use std::string::ToString;
+
+use crate::durable::upgrade::{objects_v44 as v44, objects_v45 as v45};
+
+wire_compatible!(v44::ServerConfigurationKey with v45::ServerConfigurationKey);
+wire_compatible!(v44::ServerConfigurationValue with v45::ServerConfigurationValue);
+wire_compatible!(v44::ConfigKey with v45::ConfigKey);
+wire_compatible!(v44::ConfigValue with v45::ConfigValue);
+
+const SYSTEM_CONFIGURATION_COLLECTION: TypedCollection<
+    v44::ServerConfigurationKey,
+    v44::ServerConfigurationValue,
+> = TypedCollection::new("system_configuration");
+const CONFIG_COLLECTION: TypedCollection<v44::ConfigKey, v44::ConfigValue> =
+    TypedCollection::new("config");
+
+const SYSTEM_CONFIG_KEY: &str = "config_has_synced_once";
+const CONFIG_KEY: &str = "system_config_synced";
+
+/// Migrate LD sync flag from the System Configuration Collection to the Config Collection.
```

Comment:
Operationally it was easier keep this in the configuration collection.
Can you briefly summarize the reasons to move this before merging the PR?
Also, [the troubleshooting docs](https://www.notion.so/materialize/Troubleshooting-failing-LaunchDarkly-synchronization-006583554b714dea82880338d7375907?pvs=4#20908feae9354b309c39bb4269ed904b
) will need to be updated.